avro/src/lib.rs

```diff
@@ -896,7 +896,7 @@ pub use codec::{Codec, DeflateSettings};
 pub use de::from_value;
 pub use decimal::Decimal;
 pub use duration::{Days, Duration, Millis, Months};
-pub use error::Error;
+pub use error::{Details, Error};
```

Comment:
It is not very clear that `apache_avro::Details` is about Error's details.
IMO we should either export it as `apache_avro::ErrorDetails` or not export it at all and the end users should use it as `apache_avro::error::Details`. I think I like the latter better.

Reply:
> but I think it's supposed to be public anyway

Yes, I think it should be!